src/main/java/com/epam/aidial/cfg/dao/audit/repository/AdapterHistoryRepository.java

```diff
@@ -17,11 +19,15 @@
 @LogExecution
 public class AdapterHistoryRepository extends RevisionRepository {
 
+    private final ModelJpaRepository modelJpaRepository;
     private final AdapterJpaRepository adapterJpaRepository;
     private final AdapterEntityMapper adapterEntityMapper;
 
-
     public void rollbackAdapters(Number revision, AuditReader auditReader) {
+        Iterable<ModelEntity> models = modelJpaRepository.findAll();
+        models.forEach(entity -> entity.setAdapter(null));
+        modelJpaRepository.saveAllAndFlush(models);
```

Comment:
does it produce extra record in audit table?

Reply:
can become a bug. don't we have another solutions with acceptable cost?

Reply:
We have similar logic in ApplicationTypeSchemaHistoryRepository & InterceptorRunnerHistoryRepository. Can you please create a separate issue for this describing how it can become a bug?